bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/i18n/LocaleProvider.java

```diff
@@ -25,4 +28,10 @@
      */
     Locale getLocale();
 
+    /**
+     * Provides access to the location of the installation
+     *
+     * @return location of the current installation or null if the location is not set
+     */
+    PointType getLocation();
```

Comment:
> E.g. I18nProviderImpl implements the LocaleProvider, TranslationProvider and LocationProvider interface and is configured using the service pid org.eclipse.smarthome.i18n?

Sounds like a reasonable compromise. I am just afraid that we cannot easily change `I18nProvider` to `TranslationProvider` as it is used almost everywhere. Would you dare such a refactoring?

> Hopefully we never stop to discuss the design of the software / framework.

Small English lesson: "never stop to discuss" and "never stop discussing" have a very different meaning - and I hope you actually meant the second one!